contracts/connection-router/src/state.rs

```diff
@@ -246,49 +245,319 @@ impl TryFrom<String> for Address {
     }
 }
 
+#[cw_serde]
+#[serde(try_from = "String")]
+pub struct MessageId(String);
+
+impl FromStr for MessageId {
+    type Err = ContractError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // todo: should split in exactly 2 parts when migrated to state::NewMessage
+        let split: Vec<_> = s.split(ID_SEPARATOR).filter(|s| !s.is_empty()).collect();
+        if split.len() < 2 {
+            return Err(ContractError::InvalidMessageId);
+        }
+        Ok(MessageId(s.to_lowercase()))
+    }
+}
+
+impl TryFrom<String> for MessageId {
+    type Error = ContractError;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        value.as_str().parse()
+    }
+}
+
+impl From<MessageId> for String {
+    fn from(d: MessageId) -> Self {
+        d.0
+    }
+}
+
+impl Deref for MessageId {
+    type Target = String;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Display for MessageId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl<'a> PrimaryKey<'a> for MessageId {
+    type Prefix = ();
+    type SubPrefix = ();
+    type Suffix = Self;
+    type SuperSuffix = Self;
+
+    fn key(&self) -> Vec<Key> {
+        vec![Key::Ref(self.0.as_bytes())]
+    }
+}
+
+impl KeyDeserialize for MessageId {
+    type Output = Self;
+
+    fn from_vec(value: Vec<u8>) -> StdResult<Self> {
+        let value = String::from_utf8(value).map_err(StdError::invalid_utf8)?;
+        Ok(Self(value))
+    }
+}
+
+#[cw_serde]
+pub struct CrossChainUid {
+    pub chain: ChainName,
```

Comment:
```suggestion
    pub src_chain: ChainName,
```

Reply:
I don't think "src" needs to be there. The `CrossChainUid` does not have any notion of src or dest, but just about where the msg was initiated.

Reply:
It is the source chain though. When we need the source chain, we need to get it from here, since we removed `source_chain` from the encapsulating struct. Makes it clearer and easier to understand to just call it `source_chain` or `src_chain`.

Reply:
not sure where I land on this yet. It can be a bit confusing where to get the source chain from if it's not called that. On the other hand the id doesn't really have a notion of source, as Sammy pointed out. So it's kind of weird to to name it `source_chain`. I'd prefer to have another look at the naming once the code restructuring is completed